### PR TITLE
Add CSV serializer

### PR DESCRIPTION
**Pull request by carlinmack**
:heart: Thank you for your contribution!

related to:

-  fixes https://github.com/zenodo/rdm-project/issues/633

### Description

> Please describe briefly your pull request.

Using logic from [invenio_records_rest](https://github.com/inveniosoftware/invenio-records-rest/blob/master/invenio_records_rest/serializers/csv.py) but with changes to make it compatible with this repository

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [ ] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [ ] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [ ] I've added relevant test cases.
- [ ] I've added relevant documentation.
- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/best-practices/i18n/) (for relevant code).
- [ ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
- [ ] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
- [ ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [ ] I've NOT included third-party code (copy/pasted source code or new dependencies).

**Third-party code**

If you've added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

File: tests/test_serializers.py [hunk 1]
```diff
@@ -75,36 +75,66 @@ def test_prettyprint():
...
+        format_serializer_cls=CSVSerializer,
+        object_schema_cls=UITestSchema,
+        list_schema_cls=BaseListSchema,
+    )
+    test = {"test": _("test")}
+    list_test = {"hits": {"hits": [{"test": _("test1")}, {"test": _("test2")}]}}
+    assert serializer.serialize_object(test) == "\r\n".join(["title_l10n", "test", ""])
+    assert serializer.serialize_object_list(list_test) == "\r\n".join(
+        ["title_l10n", "test1", "test2", ""]
+    )
```
Comment (carlinmack) [inline tests/test_serializers.py:123]: minor: would be nice to deserialize and compare to ensure that this is a valid CSV

File: flask_resources/serializers/csv.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        """Initialize CSVSerializer.
+
+        :param csv_excluded_fields: list of paths of the fields that
+                                    should be excluded from the final output
+        :param csv_included_fields: list of paths of the only fields that
+                                    should be included in the final output
+        :param header_separator: separator that should be used when flattening
+                                 nested dictionary keys
+        :param collapse_lists: prevent lists being expanded into many columns
+                               and instead newline seperated fields
```
Comment (carlinmack) [inline flask_resources/serializers/csv.py:48]: minor: we should test these

File: flask_resources/serializers/csv.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+                    items.extend(self._flatten(v, new_key).items())
+        elif isinstance(value, list):
+            if not self.collapse_lists:
+                for index, item in enumerate(value):
+                    # for lists, build a key with an index, e.g. title_0_subtitle
+                    new_key = parent_key + sep + str(index)
+                    if self.is_field_included(new_key):
+                        items.extend(self._flatten(item, new_key).items())
+            else:
+                # for non-collapsed lists do not include index, e.g. title_subtitle
```
Comment (carlinmack) [inline flask_resources/serializers/csv.py:122]:
```suggestion
                # for collapsed lists do not include index, e.g. title_subtitle
```

**Review by carlinmack — COMMENTED** [review]
Add generic input that showcases the parameters and how they function

-- @zzacharo

File: flask_resources/serializers/csv.py (hunk 2 above)
Comment (slint) [reply flask_resources/serializers/csv.py:48]: https://github.com/inveniosoftware/flask-resources/issues/125

**Pull request merged**